tests/integration/test_vulnerability_detector/test_cpe_helper/test_cpe_helper_fields.py

```diff
@@ -0,0 +1,300 @@
+TEST_CASES_PATH = os.path.join(TEST_DATA_PATH, 'test_cases')
+TEST_FEEDS_PATH = os.path.join(os.path.dirname(os.path.realpath(__file__)), '..', 'data', 'feeds')
+TEST_PACKAGES_PATH = os.path.join(os.path.dirname(os.path.realpath(__file__)), '..', 'data', 'vulnerable_packages')
+
+# Configuration and cases data
+t1_configurations_path = os.path.join(CONFIGURATIONS_PATH, 'configuration_cpe_indexing_wrong_tags.yaml')
+t1_cases_path = os.path.join(TEST_CASES_PATH, 'cases_cpe_indexing_wrong_tags.yaml')
+t2_configurations_path = os.path.join(CONFIGURATIONS_PATH, 'configuration_cpe_indexing_wrong_values.yaml')
+t2_cases_path = os.path.join(TEST_CASES_PATH, 'cases_cpe_indexing_wrong_values.yaml')
+
+# Custom paths
+custom_nvd_json_feed_path = os.path.join(TEST_FEEDS_PATH, 'nvd', vd.CUSTOM_NVD_FEED)
+custom_cpe_helper_path = os.path.join(TEST_FEEDS_PATH, 'cpe_helper', vd.CUSTOM_GENERIC_CPE_HELPER)
+custom_vulnerable_packages_path = os.path.join(TEST_PACKAGES_PATH, vd.CUSTOM_VULNERABLE_PACKAGES)
+
+# CPE indexing packages test configurations (t1)
+t1_configuration_parameters, t1_configuration_metadata, t1_test_case_ids = get_test_cases_data(t1_cases_path)
+t1_configurations = load_configuration_template(t1_configurations_path, t1_configuration_parameters,
+                                                t1_configuration_metadata)
+t1_systems = [metadata['system'] for metadata in t1_configuration_metadata]
+
+# CPE indexing packages test configurations (t2)
+t2_configuration_parameters, t2_configuration_metadata, t2_test_case_ids = get_test_cases_data(t2_cases_path)
+t2_configurations = load_configuration_template(t2_configurations_path, t2_configuration_parameters,
+                                                t2_configuration_metadata)
+t2_systems = [metadata['system'] for metadata in t2_configuration_metadata]
+
+# Set offline custom feeds configuration
+t1_configurations = update_configuration_template(t1_configurations, ['CUSTOM_NVD_JSON_PATH'],
+                                                  [custom_nvd_json_feed_path])
+t2_configurations = update_configuration_template(t2_configurations, ['CUSTOM_NVD_JSON_PATH'],
+                                                  [custom_nvd_json_feed_path])
+
+# Global vars
+agent_packages = read_json_file(custom_vulnerable_packages_path)
+
+
+def replace_cpe_json_fields(tags=None, values=None):
+    """Replace the tags and values of the generic_custom_cpe_helper.json file.
+
+    Args:
+        tags (dict): Dictionary with tags names values
+        values (dict): Dictionary with tag values
+    """
+    with open(CPE_HELPER_PATH, 'r') as file:
+        filedata = file.read()
+    for key, value in tags.items():
+        filedata = filedata.replace(key, value)
+    for key, value in values.items():
+        filedata = filedata.replace(key, value)
+    with open(CPE_HELPER_PATH, 'w') as file:
+        file.write(filedata)
+
+
+@pytest.fixture(scope='function')
```

Comment:
This fixture is almost the same in `test_cpe_helper_fields` and `test_cpe_helper_missing_fields`. Maybe it could be a good idea to refactor them into a common `test_cpe_helper`

Reply:
Done [here](https://github.com/wazuh/wazuh-qa/pull/3731/commits/e6b49a5a140e4b5100cf848d49091abf9eb31df6)